### PR TITLE
fix(swarm): Make executors pub and add convenience constructors for SwarmBuilder

### PR DESCRIPTION
**Pull request by umgefahren**
## Description

Addresses issues raised here: https://github.com/libp2p/rust-libp2p/pull/3097#discussion_r1026526312

## Links to any relevant issues

Follows from #3097.

## Change checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Comment by dignifiedquire** [comment]
thank you!

**Review by thomaseizinger — APPROVED** [review]

**Comment by thomaseizinger** [comment]
I'd like to merge https://github.com/libp2p/test-plans/pull/77 first so we have a green CI.

File: swarm/src/executor.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//! Provides executors for spawning background tasks.
```
Comment (mxinden) [inline swarm/src/executor.rs:1]: Oversight on my end on https://github.com/libp2p/rust-libp2p/pull/3097: Would you mind adding a license header here? See other files in the repository.

**Review by mxinden — APPROVED** [review]
As always, very much appreciate the help @umgefahren!

**Pull request merged**